include/zephyr/drivers/video.h

```diff
@@ -753,104 +753,184 @@ struct video_buffer *video_buffer_alloc(size_t size);
+	((uint32_t)(a) | ((uint32_t)(b) << 8) | ((uint32_t)(c) << 16) | ((uint32_t)(d) << 24))
+
+/**
+ * @brief Convert a four-character string to a four-character-code
+ *
+ * This converts a string literal or variable into a four-character-code
+ * as defined by @ref VIDEO_FOURCC.
+ *
+ * @param str String to be converted
+ * @return Four-character-code.
+ */
+#define VIDEO_FOURCC_FROM_STR(str) VIDEO_FOURCC((str)[0], (str)[1], (str)[2], (str)[3])
+
+/**
+ * @name Bayer formats (R, G, B channels).
  * @{
  */
 
-/** BGGR8 pixel format */
-#define VIDEO_PIX_FMT_BGGR8 video_fourcc('B', 'G', 'G', 'R') /*  8  BGBG.. GRGR.. */
-/** GBRG8 pixel format */
-#define VIDEO_PIX_FMT_GBRG8 video_fourcc('G', 'B', 'R', 'G') /*  8  GBGB.. RGRG.. */
-/** GRBG8 pixel format */
-#define VIDEO_PIX_FMT_GRBG8 video_fourcc('G', 'R', 'B', 'G') /*  8  GRGR.. BGBG.. */
-/** RGGB8 pixel format */
-#define VIDEO_PIX_FMT_RGGB8 video_fourcc('R', 'G', 'G', 'B') /*  8  RGRG.. GBGB.. */
+/**
+ * @verbatim
+ * | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | ...
+ * | Gggggggg | Rrrrrrrr | Gggggggg | Rrrrrrrr | Gggggggg | Rrrrrrrr | ...
+ * @endverbatim
+ */
+#define VIDEO_PIX_FMT_BGGR8      VIDEO_FOURCC('B', 'A', '8', '1')
+#define VIDEO_PIX_FMT_BGGR8_BITS 8
+
+/**
+ * @verbatim
+ * | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | ...
+ * | Rrrrrrrr | Gggggggg | Rrrrrrrr | Gggggggg | Rrrrrrrr | Gggggggg | ...
+ * @endverbatim
+ */
+#define VIDEO_PIX_FMT_GBRG8      VIDEO_FOURCC('G', 'B', 'R', 'G')
+#define VIDEO_PIX_FMT_GBRG8_BITS 8
+
+/**
+ * @verbatim
+ * | Gggggggg | Rrrrrrrr | Gggggggg | Rrrrrrrr | Gggggggg | Rrrrrrrr | ...
+ * | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | ...
+ * @endverbatim
+ */
+#define VIDEO_PIX_FMT_GRBG8      VIDEO_FOURCC('G', 'R', 'B', 'G')
+#define VIDEO_PIX_FMT_GRBG8_BITS 8
+
+/**
+ * @verbatim
+ * | Rrrrrrrr | Gggggggg | Rrrrrrrr | Gggggggg | Rrrrrrrr | Gggggggg | ...
+ * | Gggggggg | Bbbbbbbb | Gggggggg | Bbbbbbbb | Gggggggg | Bbbbbbbb | ...
+ * @endverbatim
+ */
+#define VIDEO_PIX_FMT_RGGB8      VIDEO_FOURCC('R', 'G', 'G', 'B')
+#define VIDEO_PIX_FMT_RGGB8_BITS 8
 
 /**
  * @}
  */
 
 /**
  * @name RGB formats
+ * Per-color (R, G, B) channels.
  * @{
  */
 
-/** RGB565 pixel format */
-#define VIDEO_PIX_FMT_RGB565 video_fourcc('R', 'G', 'B', 'P') /* 16  RGB-5-6-5 */
+/**
+ * 5-bit blue followed by 6-bit green followed by 5-bit red, in little-endian over two bytes.
+ * @verbatim
+ * | gggRrrrr | BbbbbGgg | ...
+ * @endverbatim
+ */
+#define VIDEO_PIX_FMT_RGB565      VIDEO_FOURCC('R', 'G', 'B', 'P')
+#define VIDEO_PIX_FMT_RGB565_BITS 16
 
-/** XRGB32 pixel format */
-#define VIDEO_PIX_FMT_XRGB32 video_fourcc('B', 'X', '2', '4') /* 32  XRGB-8-8-8-8 */
+/**
+ * There is an empty (X) byte for each pixel.
+ * @verbatim
+ * | Xxxxxxxx Rrrrrrrr Gggggggg Bbbbbbbb | ...
+ * @endverbatim
+ */
+#define VIDEO_PIX_FMT_XRGB32      VIDEO_FOURCC('B', 'X', '2', '4')
+#define VIDEO_PIX_FMT_XRGB32_BITS 32
```

Comment:
I understand that you want to help developers who don't need to know each pixel format is of how many bits. But I don't think this is a good idead to define each macro for each format bits. There **will** be many formats and several of them will have the same number of bits, it seems we use too many macros to repeats some numbers (8, 10, 32, etc.).

Another drawback is developer needs to remember the macro name when they can just type, for example 32.

Reply:
The logical grouping chosen by `video.h` does not match the grouping in the switch (grouped by number of bits).

It was mostly for facilitating maintenance and avoid mistakes to have each format documented at only one place.
The `switch (pixfmt) { case ...: }` implementation used here was generated to avoid mistakes, and the compiler would likely deduplicate the similar entries.

This does absolutely not matter though, and I am removing the extra macros. :)